### PR TITLE
DOC-7458: New advanced query settings

### PR DESCRIPTION
**Pull request by simon-dew**
The following draft documentation is ready for review:

[Settings and Parameters](https://simon-dew.github.io/docs-site/DOC-7458/server/7.0/settings/query-settings.html)

- Added the following request-level settings: `durability_level`, `durability_timeout`, `memory_quota`, `query_context`, `txdata`, `txid`, `tximplicit`, `txstmtnum`, `txtimeout`, `use_cbo`

- Updated the following request-level settings: `scan_consistency`, `timeout`

- Added the following service-level parameters:  `functions-limit`, `memory-quota`, `txtimeout`, `use-cbo`

- Updated the following service-level parameters: `n1ql-feat-ctrl `

Docs issue: [DOC-7458](https://issues.couchbase.com/browse/DOC-7458)

### CODE REVIEW

**Comment by simon-dew** [comment]
Upstream pull request for service-level parameters is [cb-swagger#47](https://github.com/couchbaselabs/cb-swagger/pull/47)

**Comment by marcogrecopriolo** [comment]
All the new cluster level parameters (MB-39729) are missing, but I guess they are not implemented yet.
Re encoded_plan, it is ignored in most cases, but there's a backdoor way of enabling it in case of emergencies for old clients: I'm guessing we don't want to document that, but you should be aware that it can be used.
Re the namespace example I find the use of `travel-sample` confusing, since we have a `travel-sample` bucket: indeed if the user set any namespace bar `default` no query with no query context and unqualified keyspaces would work.
Not sure how we want to improve on that - but I think we should avoid having a user try that \set command and then go WTH?

+1 from me, I'll defer to other reviewers for tx parameters.

**Review by sitaramv — COMMENTED** [review]
remove  durability_timeout it is not used

File: modules/settings/pages/query-settings.adoc [hunk 1]
```diff
@@ -1025,6 +1025,13 @@ $ curl http://localhost:8093/query/service -u user:pword -d 'statement=select *
 ----
 | string (duration)
 
+| [[txdata]]
+**txdata** +
+__Optional__
+| Transaction data.
+For internal use only.
+| string
```
Comment (sitaramv) [inline modules/settings/pages/query-settings.adoc]: JSON object

File: modules/settings/pages/query-settings.adoc [hunk 2]
```diff
@@ -1032,6 +1032,30 @@ __Optional__
 For internal use only.
 | string
 
+| [[txid]]
+**txid** +
+__Required__ for statements within a transaction
+| Transaction ID.
+Specifies the transaction to which a statement belongs.
+
+The transaction ID should be the same as the transaction ID generated by the `BEGIN TRANSACTION` statement.
```
Comment (sitaramv) [inline modules/settings/pages/query-settings.adoc:1061]: consistent with begin work, start transaction same if use transaction use same on rollback/commit

File: modules/settings/pages/query-settings.adoc [hunk 3]
```diff
@@ -1085,6 +1085,32 @@ $ curl http://localhost:8093/query/service \
...
 | boolean
 
+| [[txstmtnum]]
+**txstmtnum** +
+__Optional__
+| Transaction statement number.
+Specifies the order in which statements in a transaction should be executed.
+
+The transaction statement number must be a positive integer, and must be higher than any previous transaction statement numbers in the transaction.
+When the transaction is committed, statements are executed in ascending order of transaction statement number.
```
Comment (sitaramv) [inline modules/settings/pages/query-settings.adoc]: If the number is lower than previous statement it returns error.  remove these lines

File: modules/settings/pages/query-settings.adoc [hunk 4]
```diff
@@ -1111,6 +1111,43 @@ $ curl http://localhost:8093/query/service \
...
 | integer (int32)
 
+| [[txtimeout_req]]
+**txtimeout** +
+__Optional__
+| Maximum time to spend on a transaction before timing out.
+Only applies to `BEGIN TRANSACTION` statements, or DML statements for which {tximplicit}[tximplicit] is set.
+For other statements, it is ignored.
+
+Specify a duration of `0` to disable.
```
Comment (sitaramv) [inline modules/settings/pages/query-settings.adoc:1142]: no disable by default 2m

Comment (sitaramv) [reply modules/settings/pages/query-settings.adoc:1142]: that is request level if set 0 means 2m

File: modules/settings/pages/query-settings.adoc [hunk 5]
```diff
@@ -1111,6 +1111,43 @@ $ curl http://localhost:8093/query/service \
...
+When disabled, no timeout is applied and the transaction runs for however long it takes.
+
+The {txtimeout-srv}[node-level] `txtimeout` setting specifies the default for this parameter for a single node.
+The request-level parameter overrides the node-level setting.
+However, if the node-level setting is greater than 0, the transaction timeout for the query is limited to the node-level setting.
+
+The request-level `txtimeout` parameter is a string.
+Its format includes an amount and a mandatory unit, e.g. `10ms` (10 milliseconds) or `0.5s` (half a second).
```
Comment (sitaramv) [inline modules/settings/pages/query-settings.adoc:1150]: with in the transaction  statement level timeout setting is ignored inherits remaining time from txtimeout.
txtimeout is wall clock. The clock starts when BEGIN WORK successful. On heavily loaded system statements waiting in the queue for resources count to wards this time. once txtimeout reached no statement is allowed to continue in the transaction.

**Comment by simon-dew** [comment]
> Re the namespace example I find the use of `travel-sample` confusing, since we have a `travel-sample` bucket: indeed if the user set any namespace bar `default` no query with no query context and unqualified keyspaces would work.

Done

**Comment by simon-dew** [comment]
> remove durability_timeout it is not used

Done

File: modules/settings/pages/query-settings.adoc (hunk 1 above)
Comment (simon-dew) [reply modules/settings/pages/query-settings.adoc]: Done

File: modules/settings/pages/query-settings.adoc (hunk 2 above)
Comment (simon-dew) [reply modules/settings/pages/query-settings.adoc:1061]: Done

File: modules/settings/pages/query-settings.adoc (hunk 3 above)
Comment (simon-dew) [reply modules/settings/pages/query-settings.adoc]: Done

File: modules/settings/pages/query-settings.adoc (hunk 4 above)
Comment (simon-dew) [reply modules/settings/pages/query-settings.adoc:1142]: Done

File: modules/settings/pages/query-settings.adoc (hunk 5 above)
Comment (simon-dew) [reply modules/settings/pages/query-settings.adoc:1150]: Done

**Pull request merged**